### DISCUSSION TITLE
Auto-Assignment Policy Configuration

### DISCUSSION THREAD

**Discussion by salty2011** [ATVWGS/tenant-management-framework#13] [Ideas]
Microsoft have just GA'd the ability to configure Auto-Assignment policies on Access Packages and there is also a corresponding API that can be used to define these policies.

[Auto-Assignment Policies](https://docs.microsoft.com/en-us/azure/active-directory/governance/entitlement-management-access-package-auto-assignment-policy)

[accessPackageAutomaticRequestSettings resource type](https://docs.microsoft.com/en-us/graph/api/resources/accesspackageautomaticrequestsettings?view=graph-rest-1.0&preserve-view=truel)

Any chance we can do a feature update to include defining these policy types in TMF?

**Comment by ATVWGS** [comment]
The feature request was added to our backlog.

**Reply by salty2011** [reply]
Hi @ATVWGS any update on implementing this feature?

**Reply by brine0m** [reply]
Hey @ATVWGS 

Has any progress occurred internally on this capability?

**Reply by ATVWGS** [reply]
Hi @brine0m @salty2011,

unfortunately I had some other priorities in the last months, but I will start working on new features in TMF in March.

**Reply by ATVWGS** [reply]
Hi @brine0m @salty2011,

I'm already working on the feature. Turned out that this functionality is only available in the V1.0 endpoint and that differs widely from the beta endpoint that we used so far for assignmentPolicies. So I have to re-develop all the functions for assignmentPolicies and this will also lead to the necessity to change the configuration files to the format of the V1.0 endpoint!

**Reply by ATVWGS** [reply]
Hi @brine0m @salty2011,

I have finished the development and testing of moving to the v1.0 endpoint for assignment policies including auto assignment policies. As this change also requires to change all existing config files for accessPackages I'm not quite sure how to announce this, so that everybody who updates to the new prerelease version is aware of that. Any ideas?

**Reply by salty2011** [reply]
Hi @ATVWGS 

Apologies for the delay in reply, 

Have a few questions regarding switch back to the 1.0 API.

- Is the switch to 1.0 solely for use in access package assignment policies or across the whole module
- What functionality are you losing by this switch

The only pro I can see for 1.0 aside from supporting the auto assignment policies is stability. In theory the 1.0 API doesn't change to much which would lead to less overhead longer term I guess.

**Reply by ATVWGS** [reply]
Hi @salty2011,

the switch to the 1.0 API is solely for the access package assignment policies, because it's not available in the beta endpoint. For all other resources still the beta endpoints are used, so there's no loss on functionality at all. 

You're right that moving to the 1.0 API would lead to more stability and we already have a task for that in the backlog. We have to check which functionalities have been also made available in the 1.0 API in the meantime and if the endpoints differ in parameter sets or not.

**Comment by ATVWGS** [comment]
@brine0m @salty2011 
Functionality is added in "TMF 1.6.2238-preview". Added WIKI page "Breaking Changes" to announce code changes that needs to update existing configurations.
